pyccel/ast/cupyext.py

```diff
@@ -436,6 +442,39 @@ def __str__(self):
 
 #==============================================================================
 
+class CupyRavel(CupyArray):
+    """
+    Class representing a call to the cupy ravel function which
+    returns flattened version of the passed array
+
+    Parameters
+    ==========
+    arg : PyccelAstNode
+            A PyccelAstNode of unknown shape
+    memory_location : string
+            The location where the new array memory should be allocated
+    """
+    name   = 'ravel'
+    __slots__ = ('_arg','_dtype','_precision','_shape','_rank','_order', '_memory_location')
```

Comment:
See https://towardsdatascience.com/understand-slots-in-python-e3081ef5196d for more details about `__slots__`. There is a section on inheritance.

`__slots__` are inherited. By repeating them you create multiple objects `CupyArray.arg` and `CupyRavel.arg` in the same object. This tends to break inheritance as line 470 sets `self(CupyRavel)._shape` not `self(CupyArray)._shape` so functions in `CupyArray` don't have access to this value.

```suggestion
    __slots__ = ()
```